internal/service/device.go

```diff
@@ -393,5 +407,33 @@ func (h *ServiceHandler) DecommissionDevice(ctx context.Context, request server.
 	if !allowed {
 		return server.DecommissionDevice403JSONResponse{Message: Forbidden}, nil
 	}
-	return nil, fmt.Errorf("not yet implemented")
+
+	orgId := store.NullOrgId
+	var updateCallback func(before *model.Device, after *model.Device)
+
+	if h.callbackManager != nil {
+		updateCallback = h.callbackManager.DeviceUpdatedCallback
+	}
+	newObj := &v1alpha1.Device{
+		Metadata: v1alpha1.ObjectMeta{
+			Name: &request.Name,
+		},
+		Spec: &v1alpha1.DeviceSpec{
+			DecommissionRequested: request.Body,
+		},
+	}
+
+	// set the fromAPI bool to 'false', otherwise updating the spec.decommissionRequested of a device is blocked
+	result, err := h.store.Device().Update(ctx, orgId, newObj, nil, false, updateCallback)
```

Comment:
Doesn't this erase all other `.spec` fields instead of just adding the `decommissionRequested` field?

Reply:
True; fixed.